protovalidate/internal/constraints.py

```diff
@@ -347,8 +347,10 @@ def __init__(
         self._ignore_empty = (
             field_level.ignore_empty
             or field_level.ignore in (validate_pb2.IGNORE_IF_UNPOPULATED, validate_pb2.IGNORE_IF_DEFAULT_VALUE)
-            or field.has_presence  # type: ignore[attr-defined]
-            and not for_items
+            or (
+                field.has_presence  # type: ignore[attr-defined]
+                and not for_items
+            )
```

Comment:
@jchadwick-buf semantically this shouldn't have changed, just confirming this is the intended grouping here?

Reply:
Yes sounds right. The basic idea is that if a field has presence, then it should default to being ignored when empty. However, `for_items` is set if we're running against a value nested under a repeated field. So if `for_items` is true, then we don't want to ignore empty on the zero value: the field is definitely present.